lib/inspec/base_cli.rb

```
@@ -71,24 +73,70 @@ def self.exec_options
         desc: 'Write out a lockfile based on this execution (unless one already exists)'
       option :backend_cache, type: :boolean,
         desc: 'Allow caching for backend command output.'
+      option :show_progress, type: :boolean,
+        desc: 'Show progress while executing tests.'
     end
 
     def self.default_options
       {
         exec: {
+          'reporter' => {
+            'cli' => nil,
+          },
+          'show_progress' => false,
           'color' => true,
           'create_lockfile' => true,
           'backend_cache' => false,
         },
       }
     end
 
+    def self.clean_reporters(opts)
+      # Parse report options from CLI
+      if opts['reporter'].is_a?(Array)
+        stdout = 0
+        reports = {}
+        opts['reporter'].each do |report|
```

This is not a safe implementation. I can easily break it with:

```
{ 
    "backend_cache": true,
    "reporter": { 
        "junit" : null,
        "cli" : null
    }
}
```
And then calling `inspec exec supermarket://dev-sec/ssh-baseline --json-config config.json `

---

Yeah, that's a cool find actually. So we can either raise an exception if `opts['reporter']` isn't an array, or we can support both Array and Hash syntax. I'd prefer the latter, and then raise if we get anything other than Array or Hash.